wgpu-core/src/device/resource.rs

```diff
@@ -1823,6 +1840,8 @@ impl<A: HalApi> Device<A> {
         Ok(())
     }
 
+    // This function expects the provided bind group layout to be resolved
+    // (not passing a duplicate) beforehand.
```

Comment:
- [ ] question(blocking): I'm not sure how to validate that these should always be original/deduplicated. How can I do that?

Reply:
It is called once and the de-duplication is done just before calling it.

Reply:
I guess I should note: calling `get_bind_group_layout` or `try_get_bind_group_layout`  is what resolves the indirection (what I call de-duplication in this context) in various pieces of the code, that's generally what you want to call to make sure you have the real BGL.